src/controllers/usersController.ts

```diff
@@ -321,3 +321,94 @@ export async function updateSecondaryEmailForUser(req, res) {
     res.redirect(`/community/${username}`);
   }
 }
+
+function createBranchName(username) {
+  const refRegex = /( |\.|\\|~|^|:|\?|\*|\[)/gm;
+  const randomSuffix = crypto.randomBytes(3).toString('hex');
+  return `author${username.replace(refRegex, '-')}-update-end-date-${randomSuffix}`;
+}
+
+async function updateAuthorGithubFile(username, changes) {
+  const branch = createBranchName(username);
+  const path = `content/_authors/${username}.md`;
+  console.log(`Début de la mise à jour de la fiche pour ${username}...`);
+
+  await utils.getGithubMasterSha()
+    .then((response) => {
+      const { sha } = response.data.object;
+      console.log('SHA du master obtenu');
+      return utils.createGithubBranch(sha, branch);
+    })
+    .then(() => {
+      console.log(`Branche ${branch} créée`);
+      return utils.getGithubFile(path, branch);
+    })
+    .then((res) => {
+      let content = Buffer.from(res.data.content, 'base64').toString('utf-8');
+      changes.forEach((change) => {
```

Comment:
Nitpick: je préfère les for ... of au lieu des forEach

Reply:
pour quelle raison ?